.circleci/config.yml

```diff
@@ -15,150 +42,206 @@ references:
+  # The artifacts are also stored and made available through circleci UI if further investigations are necessary
   build:
-    parameters:
-      is_pull_request:
-        type: boolean
-        default: false
     <<: *workdir
-    docker: # run the steps with Docker
-      - image: circleci/openjdk:8-jdk-stretch
-    resource_class: large
+    docker:
+      - image: cimg/openjdk:8.0.275
     steps:
       - *attach-workspace
       - restore_cache:
           keys:
-            - v2-dependencies-{{ checksum "pom.xml" }}
-            # fallback to using the latest cache if no exact match is found
-            - v2-dependencies-
-      - when:
-          condition: <<parameters.is_pull_request>>
-          steps:
-            - run:
-                name: mvn clean install
-                command: mvn -s .circleci/.circleci.settings.xml clean install
-      - unless:
-          condition: <<parameters.is_pull_request>>
-          steps:
-            - run:
-                name: mvn clean deploy
-                command: mvn -s .circleci/.circleci.settings.xml clean deploy
+            - v1-dependencies-{{ checksum "pom.xml" }}
+      - run: mvn -s .circleci/.circleci.settings.xml clean install
       - save_cache:
           paths:
             - ~/.m2
-          key: v2-dependencies-{{ checksum "pom.xml" }}
+          key: v1-dependencies-{{ checksum "pom.xml" }}
+      # Environment provisioning is sometimes dependent upon dependencies build during mvn clean install
+      # This step copies those dependencies and make them available for further actions,
+      # such as being used to provision a running Jahia with a set of freshly built artifacts
+      - run: mvn -s .circleci/.circleci.settings.xml dependency:copy-dependencies
       - run:
-          name: Copy artifacts
+          name: Prepare artifacts
           command: |
             mkdir /tmp/artifacts/
             cp /home/circleci/source/target/*.jar /tmp/artifacts/
       - store_artifacts:
           path: /tmp/artifacts/
       - *persist-workspace
 
+  # This job publish the artifact to nexus by calling the deploy command
+  # It is only executed when merging into master (after successful PR execution)
+  publish:
+    <<: *workdir
+    docker:
+      - image: cimg/openjdk:8.0.275
+    steps:
+      - *attach-workspace
+      - restore_cache:
+          keys:
+            - v1-dependencies-{{ checksum "pom.xml" }}
+            # fallback to using the latest cache if no exact match is found
+            - v1-dependencies-
+      - run: mvn -s .circleci/.circleci.settings.xml clean install deploy
+
   sonar:
-    parameters:
-      is_pull_request:
-        type: boolean
-        default: false
     <<: *workdir
     docker: # run the steps with Docker
-      - image: circleci/openjdk:8-jdk-stretch
-    resource_class: large
+      - image: circleci/openjdk:11-jdk-stretch
+    resource_class: medium
     steps:
       - *attach-workspace
+      - checkout
```

Comment:
Another restore_cache step is missing here to restore the Maven repository ( on v1-dependencies-{{ checksum "pom.xml" }}), otherwise all the dependencies will have to be downloaded again.